src/timesync/timesyncd-conf.c

```diff
@@ -106,11 +107,42 @@ int config_parse_servers(
 }
 
 int manager_parse_config_file(Manager *m) {
+        int r;
+
         assert(m);
 
-        return config_parse_many_nulstr(PKGSYSCONFDIR "/timesyncd.conf",
-                                        CONF_PATHS_NULSTR("systemd/timesyncd.conf.d"),
-                                        "Time\0",
-                                        config_item_perf_lookup, timesyncd_gperf_lookup,
-                                        false, m);
+        r = config_parse_many_nulstr(PKGSYSCONFDIR "/timesyncd.conf",
+                                     CONF_PATHS_NULSTR("systemd/timesyncd.conf.d"),
+                                     "Time\0",
+                                     config_item_perf_lookup, timesyncd_gperf_lookup,
+                                     false, m);
+        if (r < 0)
+                return r;
+
+        if (m->min_poll_interval_usec < 16 * USEC_PER_SEC) {
+                log_warning("Invalid MinPollIntervalSec=, ignoring.");
+                m->min_poll_interval_usec = NTP_POLL_INTERVAL_MIN_SEC * USEC_PER_SEC;
+        }
+
+        if (m->max_poll_interval_usec < m->min_poll_interval_usec) {
+                log_warning("Invalid MaxPollIntervalSec=, ignoring.");
+                m->max_poll_interval_usec = MAX(NTP_POLL_INTERVAL_MAX_SEC * USEC_PER_SEC, m->min_poll_interval_usec * 32);
```

Comment:
MAX() → MIN()

Hmm, I I just set MinPollIntervalSec to a sufficiently large value, I might get a warning here, and it'll be quite confusing too.